Xcode9b2 build failing (with Carthage)

Also happening for me.
Carthage version: **0.24.0**
Cartfile entry: **github "apple/swift-protobuf" ~> 0.9.904**
Xcode: **Version 9.0 beta 3 (9M174d)**
Swift Language Version: **Swift 3.2**

Carthage builds all other libraries successfully and then:
```
*** Building scheme "SwiftProtobuf_iOS" in SwiftProtobuf.xcodeproj
Build Failed
	Task failed with exit code 65:
	/usr/bin/xcrun xcodebuild -project /Users/rick/projects/luno/ios/app/Carthage/Checkouts/swift-protobuf/SwiftProtobuf.xcodeproj -scheme SwiftProtobuf_iOS -configuration Release -derivedDataPath /Users/rick/Library/Caches/org.carthage.CarthageKit/DerivedData/swift-protobuf/0.9.904 -sdk iphonesimulator -destination platform=iOS\ Simulator,id=D7AB629C-2FF6-4AD3-956F-0FE68D1824FE -destination-timeout 3 ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES build (launched in /Users/rick/projects/luno/ios/app/Carthage/Checkouts/swift-protobuf)

This usually indicates that project itself failed to compile. 
Please check the xcodebuild log for more details: /var/folders/xm/tkhp5mf104sflmh4sdbtwgf80000gn/T/carthage-xcodebuild.9le37j.log
```

The xcode build log ends with:
```
** BUILD FAILED **

The following build commands failed:
	CompileSwift normal i386
	CompileSwiftSources normal i386 com.apple.xcode.tools.swift.compiler
(2 failures)
```

--- a/Sources/SwiftProtobuf/HashVisitor.swift
+++ b/Sources/SwiftProtobuf/HashVisitor.swift
@@ -105,7 +105,7 @@ internal struct HashVisitor: Visitor {
     fieldType: _ProtobufMap<KeyType, ValueType>.Type,
     value: _ProtobufMap<KeyType, ValueType>.BaseType,
     fieldNumber: Int
-  ) {
+  ) throws {
     mix(fieldNumber)
     mixMap(map: value)
   }
@@ -115,7 +115,7 @@ internal struct HashVisitor: Visitor {
     fieldType: _ProtobufEnumMap<KeyType, ValueType>.Type,
     value: _ProtobufEnumMap<KeyType, ValueType>.BaseType,
     fieldNumber: Int
-  ) where ValueType.RawValue == Int {
+  ) throws where ValueType.RawValue == Int {
     mix(fieldNumber)
     mixMap(map: value)
   }
@@ -125,7 +125,7 @@ internal struct HashVisitor: Visitor {
     fieldType: _ProtobufMessageMap<KeyType, ValueType>.Type,
     value: _ProtobufMessageMap<KeyType, ValueType>.BaseType,
     fieldNumber: Int
-  ) {
+  ) throws {
     mix(fieldNumber)
     mixMap(map: value)
   }

--- a/Sources/SwiftProtobuf/SelectiveVisitor.swift
+++ b/Sources/SwiftProtobuf/SelectiveVisitor.swift
@@ -246,7 +246,7 @@ internal extension SelectiveVisitor {
     fieldType: _ProtobufMessageMap<KeyType, ValueType>.Type,
     value: _ProtobufMessageMap<KeyType, ValueType>.BaseType,
     fieldNumber: Int
-  ) {
+  ) throws {
     assert(false)
   }
 

